test/fhirtypes/Range.test.ts

```
@@ -0,0 +1,112 @@
+        high: large
+      };
+      const wrongOrder: Range = {
+        low: large,
+        high: small
+      };
+      expect(validateRange(inOrder)).toBe(true);
+      expect(validateRange(equalBound)).toBe(true);
+      expect(() => {
+        validateRange(wrongOrder);
+      }).toThrow(InvalidRangeValueError);
+    });
+    it('should require that both bounds have the same units', () => {
+      const differentUnits: Range = {
+        low: cloneDeep(small),
+        high: large
+      };
+      differentUnits.low.unit = 'UCUM#kg';
+      const missingUnit: Range = {
+        low: small,
+        high: cloneDeep(large)
+      };
+      delete missingUnit.high.unit;
+      expect(() => {
+        validateRange(differentUnits);
+      }).toThrow(UnitMismatchError);
+      expect(() => {
+        validateRange(missingUnit);
+      }).toThrow(UnitMismatchError);
+    });
+    it('should require that both bounds have the same code and system', () => {
+      const differentCode: Range = {
+        low: cloneDeep(small),
+        high: large
+      };
+      differentCode.low.code = 'kg';
+      const differentSystem: Range = {
+        low: small,
+        high: cloneDeep(large)
+      };
+      differentSystem.high.system = 'http://all.wrong';
+      const missingCode: Range = {
+        low: small,
+        high: cloneDeep(large)
+      };
+      delete missingCode.high.code;
+      const missingSystem: Range = {
+        low: cloneDeep(small),
+        high: large
+      };
+      delete missingSystem.low.system;
+      expect(() => {
+        validateRange(differentCode);
+      }).toThrow(CodeAndSystemMismatchError);
+      expect(() => {
+        validateRange(differentSystem);
+      }).toThrow(CodeAndSystemMismatchError);
+      expect(() => {
+        validateRange(missingCode);
+      }).toThrow(CodeAndSystemMismatchError);
+      expect(() => {
+        validateRange(missingSystem);
+      }).toThrow(CodeAndSystemMismatchError);
+    });
+  });
```

Probably a stylistic thing, but... you seem to test multiple things in a single `it` -- whereas I might have broken them up some more.  I also like to describe the expected behavior a bit more specifically.  As opposed to saying it `should require...` -- which is ambiguous about what happens if the requirement is broken, I like saying it `should throw...` For example, this last test I might have broken into tests like this:
* `it('should throw when the low and high have different codes', () =>{`
* `it('should throw when the low and high have different systems', () =>{`
* `it('should throw when low or high is missing a code', () =>{`
* `it('should throw when low or high is missing a system', () =>{`

This way when a test fails you can more easily pinpoint exactly what failed.  It also provides a nice set of requirements describing what should happen in what cases.